hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -122,6 +85,57 @@ public void run() {
         }
     }
 
+    private int readFrame(List<QueryTaskSource> taskSources, int maxReads) throws Exception {
```

Comment:
"readFrames" (plural) is a little closer, but the important parts that aren't included are the things like lazy activation, selecting other hosts, and injecting source names.